src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -47,8 +48,22 @@ import           ZkFold.Symbolic.Data.UInt
 -- Bit layout is Big-endian.
 --
 newtype ByteString (n :: Natural) a = ByteString [a]
-    deriving (Haskell.Show, Haskell.Eq, Generic, NFData)
-
+    deriving
+      ( Haskell.Show
+      , Haskell.Eq
+      , Generic
+      , NFData
+      , Haskell.Functor
```

Comment:
Are Functor, Foldable and Traversable used anywhere? 
With the approach I propose it won't be an instance of those anymore.
```
newtype ByteString (n :: Natural) (backend :: Natural -> Type -> Type) (a :: Type) = ByteString (backend n a)
    deriving (Haskell.Show, Haskell.Eq, Generic, NFData)
```

Reply:
So long as `backend` has those instances then so will `ByteString`. Any symbolic data structure necessarily is a known length list of field elements so ought to be `Traversable` and `VectorSpace`.

Reply:
But ArithmeticCircuit is not a Functor because of witness